src/resmom/linux/alps.c

```diff
@@ -4082,13 +4290,36 @@ inventory_to_vnodes(basil_response_t *brp)
 		ATR_TYPE_STR, aflag, NULL) == -1)
 		goto bad_vnl;
 
+	/*
+	 * Extract KNL NIDs (Node ID) from 'knl_node_list' ('knl_node_list' is a string
+	 * containing a rangelist of KNL Node IDs) and populate 'arr_nodes'.
+	 * If BASIL 1.7 is not supported on the Cray system, knl_node_list remains empty,
+	 * causing NULL to be returned and node_count to be set to 0.
+	 */
+
+	if (basil_1_7_supported)
+		arr_nodes = process_nodelist_KNL(knl_node_list, &node_count);
 	/*
 	 * now create the compute nodes
 	 */
 	inv = &brp->data.query.data.inventory;
 	for (order = 1, node = inv->nodes; node; node = node->next, order++) {
 		char		*arch;
-		static int	first_compute_node = 1;
+
+		/*
+		 * We are only interested in creating non-KNL vnodes in this function.
+		 * We avoid creating vnodes for KNL Nodes here since they will be
+		 * created in system_to_vnodes_KNL(). So, filter them out here.
+		 */
+		skip_node = 0;
+		for (idx = 0; idx < node_count; idx++) {
+			if (node->node_id == arr_nodes[idx]) {
```

Comment:
I think you need to check for arr_nodes to be NOT-NULL before dereferencing it. If system does not support basil1.7 then it will crash here

Reply:
Thanks, @arungrover for pointing out this, added a check.